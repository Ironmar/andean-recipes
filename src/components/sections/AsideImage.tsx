import React from "react";
import foodDish from "../../assets/images/food_dish.jpg";
const AsideImage: React.FC = () => {
  return (
    <div className="image-container">
      <img src={foodDish} alt="food_dish" />
    </div>
  );
};
export default AsideImage;
