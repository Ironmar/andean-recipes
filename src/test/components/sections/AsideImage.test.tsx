import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";

import "@testing-library/jest-dom";
import AsideImage from "../../../components/sections/AsideImage";
import foodDish from "../../../assets/images/food_dish.jpg";

configure({ adapter: new Adapter() });
//@ts-ignore
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Test in <AsideImage/>", () => {
  const wrapper = shallow(<AsideImage />);
  test("should be return the component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("shuold be return the correct image path", () => {
    const image = wrapper.find("img");
    expect(image.prop("src")).toEqual(foodDish);
    expect(image.prop("alt")).toEqual("food_dish");
  });
});
