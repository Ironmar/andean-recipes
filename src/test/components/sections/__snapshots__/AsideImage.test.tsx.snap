// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AsideImage/> should be return the component correctly 1`] = `
<div
  className="image-container"
>
  <img
    alt="food_dish"
    src="food_dish.jpg"
  />
</div>
`;
